doc/decisions/0013-referencing-supplementary-data.md

```diff
@@ -0,0 +1,160 @@
+
+   A new property named `selectors`, an array of strings, will be added to the value source definition. This will allow us to reference data from a nested structure. The existing property `selector`, which is a string, won't be used for supplementary data sources as that only supports a single level of nesting.
+   The use of `selectors` will only support referencing nested data of multiple objects and not arrays.
+
+   **Single level of nesting**
+
+   ```json
+   {
+     "source": "supplementary_data",
+     "identifier": "note",
+     "selectors": ["title"]
+   }
+   ```
+
+   The above would resolve to the string: `Volume of total production`
+
+   **Multiple levels of nesting**
+
+   ```json
+   {
+     "source": "supplementary_data",
+     "identifier": "note",
+     "selectors": ["example", "description"]
+   }
+   ```
+
+   The above would resolve to the string: `Any production that was incorporated into the manufacture of other products.`
+
+---
+
+### Referencing data for repeating items
+
+As documented in the `sds-schema-definitions` repo, the top-level key `items` represents repeating items, for example, a list of employees, companies, products etc. This has special meaning in Runner as any "list" of things defined in `items` will be preloaded in eQ Runner's List Store. This allows us to use any looping or repeating feature backed by a list store, such as looping 2.5, repeating sections etc.
+For example, `items.products` will result in a list named `products` being added to the list store. The identifiers will be uniquely generated by eQ. eQ will not use the identifiers defined in the supplementary data as we cannot control an ID's structure and allowed values. However, on submission, eQ will provide downstream with a mapping of list item IDs to supplementary data IDs.
+
+When referencing data for a repeating item, there will be no need to use the keys such as `"items"` in the source definition. This structure provides a generic way to define repeating items and for Runner to parse it with enough context.
+
+Referencing data for repeating items works similarly to non-repeating items. The reference will be made using the list item name.
+
+A key point to understand here is that referencing data for a repeating item will only be supported in the context of a "repeat". For example, inside the list collector child pages, such as the repeating questions, or anywhere inside a repeating section.
+This is because, in both cases, Runner will have access to the list item id of the current item, which is what Runner will use to look up the data for the correct item. This is identical to how values for **answers** are resolved inside a repeat. We use the answer ID and the list item ID to resolve the value.
+
+For both the examples below, we'll assume the list item id for the location (URL) is `abCdE`, which behind the scenes is Runner's ID which maps to the SDS ID `89929001` (`items.products.values.identifier`).
```

Comment:
Could we elaborate on how the Runner Item ID maps to SDS Item ID when referencing  value for an SDS list item? 
Above it's mentioned that EQ provides the Runner Item ID to SDS Item ID to downstream when the list item is created and submitted, how will a later survey have context of this mapping that happened in a prev survey?

Reply:
If this won't be known until implementation, that's fair, can then be left as is

Reply:
The runner ID won't be used by later surveys, it is purely for runner and for downstream to work out the instance of an answer. It tells them which answers belong to which item. Since for prepop list, it will still use list item ids, the mapping we send downstream is just so they can do a mapping to say so "answer-1" "instance-1" is for product id `89929001` etc.

> how the Runner Item ID maps to SDS Item ID when referencing value for an SDS list item?

Do you mean literally how it will map?. The sds item id won't be used for logic until it is needed for the downstream mapping, all runner lookups will be by runner item id.

Reply:
I'm with you now, sounds good 👍